src/shared/src/backend_api.rs

```diff
@@ -0,0 +1,27 @@
+//! Selected backend canister API signatures.
+#![allow(clippy::missing_errors_doc)] // The code is auto-generated.  Baybe `didc bind` will preserve comments in future.
+use crate::types::{Guards, Stats};
+use candid::{self, Principal};
+use ic_cdk::api::call::CallResult as Result;
+
+/// Client for the backend canister, implementing a subset of the API.
+///
+/// Note: The canister signatures are generated by: `didc bind --type rs src/backend/backend.did`
```

Comment:
Are you sure this is not available (at least as an experimental feature)? I remember Yan Chen giving a demo of rust code generated from candid with a lot of customization...

Reply:
I'm not sure.  I would love that. But even if it is available, it would take time to learn how to use it and to reap the full benefits here.  I would postpone such a refactoring until after the migration.  Maybe the right thing is to make a Jira ticket and revisit it when we are in less of a rush.  Better: I'll want to learn about this anyway, so I'll  take some of my own time to become familiar with the latest state.  That is a better motivation than Jira.